client/src/main/java/io/pravega/client/stream/impl/CheckpointState.java

```diff
@@ -206,7 +215,7 @@ CheckpointState copy() {
         Map<String, Map<Segment, Long>> cpps = new HashMap<>();
         checkpointPositions.forEach((cp, pos) -> cpps.put(cp, new HashMap<>(pos)));
         Map<Segment, Long> lcp = lastCheckpointPosition == null ? null : new HashMap<>(lastCheckpointPosition);
-        return new CheckpointState(cps, ucph, cpps, lcp);
+        return new CheckpointState(cps, ucph, cpps, lcp, true);
```

Comment:
If there is compaction then the `isLastCheckpointPublished` would be set to true, even though it was never published. This would be an issue right ?

Reply:
Good catch. Yes, thats' a mistake. Fixed it now.